src/controller/c/supervisor.c

```diff
@@ -2091,6 +2174,32 @@ WbFieldRef wb_supervisor_node_get_field(WbNodeRef node, const char *field_name)
   return result;
 }
 
+const int wb_supervisor_node_get_field_count(WbNodeRef node) {
+  robot_mutex_lock_step();
+  requested_field_count = true;
```

Comment:
You should also reset `node_field_count = -1` before calling `wb_robot_flush_unlocked()` to handle correctly errors is the node is not found or communication errors.

Same for `wb_supervisor_node_get_proto_field_count`.